src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -525,16 +514,25 @@ SHAMap::addKnownNode (const SHAMapNodeID& node, Blob const& rawNode,
 
             if (newNode && isInconsistentNode(newNode))
             {
-                return SHAMapAddNode::invalid();
+                state_ = SHAMapState::Invalid;
+                return SHAMapAddNode::useful();
```

Comment:
If we're trying to fetch a map, presumably we think it's valid, otherwise why would we be fetching it? If a peer sends us information that allows us to prove that the map we're trying to fetch is invalid, that's extremely useful. That's the information we most need in this situation and there's nothing better the peer could possibly do.

This is case where we requested a node by hash, the peer gave us data that hashed to the hash we requested, and that data was an invalid node. This proves the map is invalid and there's no point in further attempts to fetch it.

The `SHAMapAddNode::useful` value means the data was useful. Allowing us to determine that the map is invalid and thus we can stop trying to get it is useful.